### ISSUE TITLE
break-after:avoid on heading not honored when float exists after the heading

### ISSUE THREAD

**Issue by MurakamiShinyu** [vivliostyle/vivliostyle.js#904]
Headings should not appear at bottom of page, so `break-after: avoid` is specified on h1-h6 by default.

However, when a page float is specified just after the heading,  `break-after: avoid` does not work.  This must be a bug.

```html
  <p>Lorem ipsum…</p>
  <h2>Heading</h2>
  <figure>
    Page top float
  </figure>
  <p>Totam est ut….</p>
```

```css
figure {
  float: top;
  float-reference: page;
}
```

Test sample HTML+CSS: https://gist.githubusercontent.com/MurakamiShinyu/9642126c1166ec3261c9ed720cd35b59/raw/break-after-avoid-with-page-float.html

Test with Vivliostyle Viewer (v2.14.5):
https://vivliostyle.github.io/viewer/v2.14.5/#src=https://gist.githubusercontent.com/MurakamiShinyu/9642126c1166ec3261c9ed720cd35b59/raw/break-after-avoid-with-page-float.html

Result screenshot:

![スクリーンショット 2022-04-16 12 17 27](https://user-images.githubusercontent.com/3324737/163659870-ea51c0f2-d871-45c3-85d4-358cde056dbd.png)

**Comment by MurakamiShinyu** [comment]
This problem is also the case for normal floats. So I changed the issue title.

A test sample: https://gist.githack.com/MurakamiShinyu/ca51c3caaaddc158b262e5f47e90579a/raw/float-test.html

Test with Vivliostyle Viewer (v2.15.2):
https://vivliostyle.github.io/viewer/v2.15.2/#src=https://gist.githack.com/MurakamiShinyu/ca51c3caaaddc158b262e5f47e90579a/raw/float-test.html

![Screen Shot 2022-05-26 at 11 57 21](https://user-images.githubusercontent.com/3324737/170406489-1640b815-8379-4037-b2c1-589110736867.png)

**Comment by MurakamiShinyu** [comment]
Fixed in #931 (with followup fix #932).

Test:

- https://vivliostyle-git-fix-issue904-vivliostyle.vercel.app/#src=https://gist.githubusercontent.com/MurakamiShinyu/9642126c1166ec3261c9ed720cd35b59/raw/break-after-avoid-with-page-float.html
- https://vivliostyle-git-fix-issue904-vivliostyle.vercel.app/#src=https://gist.githack.com/MurakamiShinyu/ca51c3caaaddc158b262e5f47e90579a/raw/float-test.html
